extension/solarwindsapmsettingsextension/extension.go

```diff
@@ -76,6 +87,131 @@ func (extension *solarwindsapmSettingsExtension) Shutdown(_ context.Context) err
 }
 
 func refresh(extension *solarwindsapmSettingsExtension) {
-	// Concrete implementation will be available in later PR
-	extension.logger.Info("refresh task")
+	extension.logger.Info("Time to refresh from " + extension.config.Endpoint)
+	if hostname, err := os.Hostname(); err != nil {
+		extension.logger.Error("Unable to call os.Hostname() " + err.Error())
+	} else {
+		ctx, cancel := context.WithTimeout(context.Background(), GrpcContextDeadline)
+		defer cancel()
+
+		request := &collectorpb.SettingsRequest{
+			ApiKey: extension.config.Key,
+			Identity: &collectorpb.HostID{
+				Hostname: hostname,
+			},
+			ClientVersion: "2",
+		}
+		if response, err := extension.client.GetSettings(ctx, request); err != nil {
+			extension.logger.Error("Unable to getSettings from " + extension.config.Endpoint + " " + err.Error())
+		} else {
+			switch result := response.GetResult(); result {
+			case collectorpb.ResultCode_OK:
+				if len(response.GetWarning()) > 0 {
+					extension.logger.Warn(response.GetWarning())
+				}
+				var settings []map[string]any
+				for _, item := range response.GetSettings() {
+					marshalOptions := protojson.MarshalOptions{
+						UseEnumNumbers:  true,
+						EmitUnpopulated: true,
+					}
+					if settingBytes, err := marshalOptions.Marshal(item); err != nil {
+						extension.logger.Warn("Error to marshal setting JSON[] byte from response.GetSettings() " + err.Error())
+					} else {
+						setting := make(map[string]any)
+						if err := json.Unmarshal(settingBytes, &setting); err != nil {
+							extension.logger.Warn("Error to unmarshal setting JSON object from setting JSON[]byte " + err.Error())
+						} else {
+							if value, ok := setting["value"].(string); ok {
+								if num, err := strconv.ParseInt(value, 10, 0); err != nil {
+									extension.logger.Warn("Unable to parse value " + value + " as number " + err.Error())
+								} else {
+									setting["value"] = num
+								}
+							}
+							if timestamp, ok := setting["timestamp"].(string); ok {
+								if num, err := strconv.ParseInt(timestamp, 10, 0); err != nil {
+									extension.logger.Warn("Unable to parse timestamp " + timestamp + " as number " + err.Error())
+								} else {
+									setting["timestamp"] = num
+								}
+							}
+							if ttl, ok := setting["ttl"].(string); ok {
+								if num, err := strconv.ParseInt(ttl, 10, 0); err != nil {
+									extension.logger.Warn("Unable to parse ttl " + ttl + " as number " + err.Error())
+								} else {
+									setting["ttl"] = num
+								}
+							}
+							if _, ok := setting["flags"]; ok {
+								setting["flags"] = string(item.Flags)
+							}
+							if arguments, ok := setting["arguments"].(map[string]any); ok {
+								if value, ok := item.Arguments["BucketCapacity"]; ok {
```

Comment:
can you add code coverage for all this new code?

Reply:
Sure. Done